keps/sig-architecture/3138-increase-reliability-bar/README.md

```diff
@@ -0,0 +1,415 @@
+Given the maturity level the project is reaching and the non-negligible amount of
+technical debt accumulated over the years, the most important aspect of the
+proposal is to build the culture which prioritizes certain types of contributions
+over purely feature work.
+
+We have to acknowledge that fixing, and especially debugging, reliability or
+stability issues is the kind of work that very few people like doing - for the
+majority it’s something that they would never do if they wouldn’t have to.
+This is especially true when it comes to debugging flaky tests.
+Moreover, while in companies there are multiple ways of enforcing that certain
+types of work is actually happening (e.g. oncall rotations, dedicated sprints, etc.),
+it’s much harder to achieve in the projects driven by the open-source community.
+
+As a result, we’re proposing introducing a process that will allow us to ensure
+that this kind of work that generally attracts fewer developers  will be really
+happening. This means, that for every piece of such work, we will introduce a two
+phase approach:
+1. announce the reliability-related issue to the SIG asking for prioritizing it
+   with a realistic (and non-conservative) deadline.
+   *We really hope that this will be effective in the majority of cases and
+   we will never be forced to use the point (2) special powers.*
+1. if the issue (reported above) is not addressed before the deadline, block
+   new features from the owning SIG until the issue will get addressed
+
+
+### Risks and Mitigations
+
+**We acknowledge that it may hurt the overall productivity and velocity of the
+community in the short-term. However, this is necessary to increase both
+reliability and our velocity medium and longer term.
+
+To mitigate the risk, we will provide the necessary tooling to ensure that any
+single SIG will not get surprised by the consequences of conscious
+deprioritization of reliability-related work.**
+
+
+Another risk is that due to pulling engineers away from the most interesting
+feature work towards non-fancy work, we may see attritions and decrease in
+contributor base. This is why it is critical to actually measure the benefits
+from this effort. Most of the metrics will turn out to be lagging indicators,
+so they will rather be used to measure the impact and reflect on the process
+periodically, not necessarily guide immediate future actions.
+
+The exact metrics are to-be-decided, but the current candidates include:
+- number of bugs (ideally weighed by severity) reported per SIG for already
+  released versions:
+  - for cross-cutting issues, they will be counted towards “owning SIG”
+- number of cherrypicks per SIG for a given minor release (targeting low
+  numbers meaning that SIG doesn’t have any issues that need to be addressed).
+  - for cross-cutting fixes, they will be counted towards “owning SIG”
+- test flakiness (as observed in https://sippy.k8s.io/?release=kube-master)
```

Comment:
this should be scoped to jobs that are reliable, a quick look to this link show some jobs that are not ...

Reply:
nevermind, is answered in the next section, scratch it, sorry